Code/positionnement/testFunibot.cpp

```diff
@@ -0,0 +1,343 @@
+#include "testFunibot.h"
+#include <random>
+
+const int MAXINFO = 32;
+
+std::ostream& operator<< (std::ostream& os, FuniTest::rapport rap)
+{
+	os << "Test: " << rap.nom << "\n";
+	if (rap.reussite)
+	{
+		os << "réussite du test\n";
+	}
+	else
+	{
+		os << "échec du test\n";
+	}
+	os << rap.info;
+	os << "\n";
+	return os;
+}
+
+double rand(double min, double max)
+{
+	double delta = max - min;
+	return ((double)std::rand() / (double)RAND_MAX) * delta + min;
+}
+
+Funibot testBot()
+{
+	std::srand(102545);
+	Funibot bot;
+
+	//ajout des poles
+	bot.addPole(FuniMath::Vecteur(rand(-2,2),rand(-2,2)+100,rand(-2,2)), FuniMath::Vecteur(rand(-2, 2), rand(-2, 2), rand(-2, 2)));
+	bot.addPole(FuniMath::Vecteur(rand(-2, 2)+100, rand(-2, 2)+100, rand(-2, 2)), FuniMath::Vecteur(rand(-2, 2), rand(-2, 2), rand(-2, 2)));
+	bot.addPole(FuniMath::Vecteur(rand(-2, 2), rand(-2, 2)+100, rand(-2, 2)+100), FuniMath::Vecteur(rand(-2, 2), rand(-2, 2), rand(-2, 2)));
+	bot.addPole(FuniMath::Vecteur(rand(-2, 2)+100, rand(-2, 2)+100, rand(-2, 2)+100), FuniMath::Vecteur(rand(-2, 2), rand(-2, 2), rand(-2, 2)));
```

Comment:
Beaucoup de rand(-2, 2). Ça vaudrait la peine de mettre des constantes pour -2 et 2, ou de définir une petite fonction ou macro pour rand(-2, 2) qui crée un vecteur avec des rand(-2,2), avec un certain offset.
Mettons:
```c++
// Méthode statique de la classe FuniMath::Vecteur par exemple
FuniMath::Vecteur FuniMath::Vecteur::vecteurRandom(offsetX = 0, offsetY = 0, offsetZ = 0)
{
	return std::move(FuniMath::Vecteur(rand(-2, 2) + offestX, rand(-2, 2) + offsetY, rand(-2, 2) + offsetZ));
}
```
Tu pourrais aussi ajouter la plage du random en paramètre (avec potentiellement les valeurs -2 et 2 par défaut).

Ensuite, tu peux juste faire ça:
```c++
using FuniMath::Vecteur::vecteurRandom;
```
```suggestion
	bot.addPole(vecteurRandom(0, 100, 0), vecteurRandom());
	bot.addPole(vecteurRandom(100, 100, 0), vecteurRandom());
	bot.addPole(vecteurRandom(0, 100, 100), vecteurRandom());
	bot.addPole(vecteurRandom(100, 100, 100), vecteurRandom());
```